### PR TITLE
workaround quote escape issues for `bun run`

### LINKED ISSUE #53: Spaces and quotes aren't handled correctly by bun run
package.json:
```json
{
  "scripts": {
    "argv": "node -e 'console.log(process.argv)'"
  }
}
```

Screenshot 2021-11-05 at 1 55 05 AM

### PR DESCRIPTION
**Pull request by alexlamsl**
fixes #53

### CODE REVIEW

File: src/cli/run_command.zig [hunk 1]
```diff
@@ -192,6 +192,22 @@ pub const RunCommand = struct {
 
                     delimiter = 0;
                 },
+                // TODO: handle escape sequences properly
+                // https://github.com/oven-sh/bun/issues/53
+                '\\' => {
+                    delimiter = 0;
+
+                    switch (script[entry_i + 1]) {
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig]: does this need a bounds check for entry_i + 1 < script.len

Comment (Jarred-Sumner) [reply src/cli/run_command.zig]: `"foo": "bar\\\"`

File: test/harness.ts [hunk 2]
```diff
@@ -81,7 +81,7 @@ export function hideFromStackTrace(block: CallableFunction) {
 }
 
 export function tempDirWithFiles(basename: string, files: Record<string, string>) {
-  const dir = fs.mkdtempSync(path.join(os.tmpdir(), basename + "_"));
+  const dir = fs.mkdtempSync(path.join(fs.realpathSync(os.tmpdir()), basename + "_"));
```
Comment (Electroid) [inline test/harness.ts:84]: Looks like there are some new `test-test.test.ts` failures, might be from this change?

Comment (alexlamsl) [reply test/harness.ts:84]: 1. tests on `bun-darwin-aarch64` weren't even running before due to `sudo`, which was fixed recently: #3254
2. `test-test.test.ts` is also failing in a separate PR: https://github.com/oven-sh/bun/actions/runs/5248738253/jobs/9480987433?pr=3288#step:6:7

File: src/cli/run_command.zig (hunk 1 above)
Comment (alexlamsl) [reply src/cli/run_command.zig]: Oops &minus; while that's not valid JavaScript/JSON, we shouldn't segfault either... 😓

**Comment by Jarred-Sumner** [comment]
Thank you

**Pull request merged**